### DISCUSSION TITLE
How to prevent ligatures for some glyphs?

### DISCUSSION THREAD

**Discussion by ghost** [typst/typst#1653] [Q&A] [answered]
Here is an example that has weird ligatures that I want to prevent, but I don't know how to force the `???` and `!!!` to stay three glyphs.
I tried show rules but this didn't work.

```typst
#set text(lang: "en")
#set par(justify: true)
??? !!! #{"???"} #{"!!!"} should not have ligatures but they have some weird ligatures
```

Update: the solution is shown [here](https://github.com/typst/typst/discussions/1653#discussioncomment-6355415)

**Comment by laurmaedje** [comment, marked as answer]
Have you tried `#text(ligatures: false)[???]` or `#set text(ligatures: false)` globally?

**Reply by ghost** [reply]
Thanks for you help, I'd like to keep all other ligatures. The nearest thing would be `#show "???": text(ligatures: false)[???]` but this results in a stack overflow.

This one nearly solves it but is not in line with the other text:

```typst
#show "???": stack(dir: ltr, "?", "?", "?")
```

**Reply by ghost** [reply]
I thought about it again. Since I'm using the `typst` Python library I can replace `"???"` by `text(ligatures: false)[???]` on Python level to replace the "???" in the user data.
```typst
#set text(lang: "en")
#set par(justify: true)
#{"test " + text(ligatures: false)[???] + " test"}
```

**Reply by laurmaedje** [reply]
You can also do `#show "???": set text(ligatures: false)` which is a bit more concise and prevents the stack overflow. Medium-term, I also want to fix the stack overflow.

**Reply by ghost** [reply]
Perfect, thank you. That is exactly what I needed, works good!
